src/jbtoc.ts

```diff
@@ -137,8 +132,9 @@ async function getBookConfig(
 }
 
 async function ls(pth: string): Promise<any> {
-  const baseUrl = '/api/contents/';
-  const fullPath = `${baseUrl}${pth}?content=1`;
+  const baseUrl = `${window.location.origin}${window.location.pathname.split('/lab')[0]}`;
```

Comment:
Can we guarantee that the URL will always have '/lab/' in it? Bare array accesses like this always make me a little nervous, so I tend to wrap them in some sort of try/catch or 'if len() > 1' block

Reply:
It smells a little to me too. The change provides an explicit path to support Binder because `fetch` was prepending an incorrect URL (missing repo name and session hash).

I think the cases where the URL might not contain `/lab` would be in Classic Notebook (which wouldn't support a JupyterLab widget anyhow) or if someone did something unexpected to the configuration, at which point it would be hard to parse the correct baseURL. 

Looking at this more closely, I think that in its current state, line 135 is going to break the toc in OSL. It has 2 instances of 'lab' in its URLs, and it should split on the last not the first. 

I am going to add a separate function to get baseURLs that handles multiple and no instances of 'lab' in the URL. If there are none, I'll just return the full `window.location.origin`.